inst/scripts/insert_image.R

```diff
@@ -5,6 +5,16 @@ local({
   if (ctx$path == '') stop(
     'Please select the blog post source file before using this addin', call. = FALSE
   )
+  ctx_ext = switch(
+    tolower(xfun::file_ext(ctx$path)),
+    "rmd" = "rmd",
+    "md" = "md",
+    "rmarkdown" = "md",
+    stop(
```

Comment:
I hope that we only check for `.rmd` and use the `![](){}` syntax. Other extensions are not necessarily problematic in Hugo (e.g. Hugo supports AsciiDoc).

Reply:
So the check I added does limit the "Insert Image" addin to files with extension `rmd`, `md` or `rmarkdown`. The version in the master branch currently assumes that markdown syntax will be valid unless width/height need to be set, in which case it uses HTML.

Based on your comment, what I can do is check the file extension and make sure it's one of [the Hugo-supported source files](https://github.com/gohugoio/hugo/blob/77c60a3440806067109347d04eb5368b65ea0fe8/helpers/general.go#L67-L78). 

Then we can use `![]()` if we're in a markdown-capable source file (without width/height set), or `![](){}` if we're in an `.Rmd` file (with w/h set). Otherwise, we can use `shiny::img()` for standard HTML for any other extensions.

I'm assuming that AsciiDoc, RST pass through HTML, as the image syntax in those formats is different from markdown.

If the file is not a potential blog source file (say an R script), should we exit, warn or add some other kind of interaction (like a pop-up modal confirming the user wants to add the image to the file)?

Reply:
I mean we don't need to be strict here. If the file is .Rmd or the width/height are not set, we use the Markdown syntax. For all other cases, use HTML. If the user tries to insert an image to an R script, I'd say just let it go (without checking anything on our side, and we assume he/she knows what he/she is doing).